src/DetailsView/components/load-assessment-button.tsx

```diff
@@ -3,31 +3,125 @@
 
 import { AssessmentDataParser } from 'common/assessment-data-parser';
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { Tab } from 'common/itab';
+import {
+    AssessmentStoreData,
+    PersistedTabInfo,
+} from 'common/types/store-data/assessment-result-data';
+import { TabStoreData } from 'common/types/store-data/tab-store-data';
+import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
+import { UrlParser } from 'common/url-parser';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
-import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
+import { LoadAssessmentDialog } from 'DetailsView/components/load-assessment-dialog';
 import * as React from 'react';
 
 export type LoadAssessmentButtonDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
     assessmentDataParser: AssessmentDataParser;
-    loadAssessmentHelper: LoadAssessmentHelper;
+    urlParser: UrlParser;
+    fileReader: FileReader;
+    document: Document;
 };
 export interface LoadAssessmentButtonProps {
     deps: LoadAssessmentButtonDeps;
+    tabStoreData: TabStoreData;
+    assessmentStoreData: AssessmentStoreData;
+}
+export interface LoadAssessmentButtonState {
+    newTab: Tab;
```

Comment:
nit: I think we need to move away from the 'Tab' moniker in this kind of use-case. The idea of a tab doesn't have much to do with this and while I don't like using `Tab` the interface due to previous naming, I definitely want to avoid props with that naming convention.

Ideas:
- `fromData` and `toData`
- `previousTargetPageData' and `newTargetPageData`

Reply:
happy with this suggestion, I used previousTargetPageData and newTargetPageData